src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/CachedCounterInfo.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe
+{
+    public class CachedCounterInfo
```

Comment:
I'll suggest a few name changes, mostly related to this class:
- Rename `CachedCounterInfo` to `CounterMetadata`. This class seems to represent a set of data that is immutable for a given meter-instrument-tags identity, so calling it metadata seems to be more descriptive. The fact that caching is used is not a detail of the class itself, so I would remove that from the name.
- Rename `ICounterPayload.CounterInfo` to `ICouterPayload.Metadata`. This will conflict with the existing `Metadata` property, so...
- Rename `ICounterPayload.Metadata` to `ICounterPayload.ValueTags`. This will better describe that this property is for (it is the set of tags specific to this value) and aligns it better with the `MeterTags` and `InstrumentTags` names on this class.

@noahfalk, I saw you had opinions on naming in [this comment](https://github.com/dotnet/diagnostics/pull/4324#discussion_r1366555241), so I wanted to get your thoughts on the suggested rename.

Reply:
I took most of these, with the slight adjustment of doing `CounterMetadata` instead of `Metadata` - that felt more clear to me, and less likely to create confusion by changing what `Metadata` refers to. I also considered doing `ValueMetadata` to keep them consistently named with each other, but the current iteration uses `ValueTags` as you suggested.

Reply:
All the names you suggested sounded fine to me @jander-msft 👍